android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -2054,6 +2107,83 @@ public void setPausedModifier(boolean paused) {
         }
     }
 
+    public void setEnterPictureInPictureOnLeave(boolean enterPictureInPictureOnLeave) {
+        this.enterPictureInPictureOnLeave = Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && enterPictureInPictureOnLeave;
+    }
+
+    protected void setIsInPictureInPicture(boolean isInPictureInPicture) {
+        this.isInPictureInPicture = isInPictureInPicture;
+        eventEmitter.onPictureInPictureStatusChanged(isInPictureInPicture);
+
+        if (fullScreenPlayerView != null && fullScreenPlayerView.isShowing()) {
+            if (isInPictureInPicture) fullScreenPlayerView.hideWithoutPlayer();
+            return;
+        }
+
+        Activity currentActivity = themedReactContext.getCurrentActivity();
+        if (currentActivity == null) return;
+
+        View decorView = currentActivity.getWindow().getDecorView();
+        ViewGroup rootView = decorView.findViewById(android.R.id.content);
+
+        LayoutParams layoutParams = new LayoutParams(
+                LayoutParams.MATCH_PARENT,
+                LayoutParams.MATCH_PARENT);
+
+        if (isInPictureInPicture) {
+            ((ViewGroup)exoPlayerView.getParent()).removeView(exoPlayerView);
```

Comment:
In production, in some cases with have following crash

<details>
<summary>06-14 15:07:04.065 32493 32493 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.view.ViewGroup.removeView(android.view.View)' on a null object reference</summary>
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at com.brentvatne.exoplayer.ReactExoplayerView.setIsInPictureInPicture(ReactExoplayerView.java:2132)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at com.brentvatne.exoplayer.ReactExoplayerFragment.onPictureInPictureModeChanged(ReactExoplayerFragment.kt:35)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at androidx.fragment.app.Fragment.performPictureInPictureModeChanged(Fragment.java:3221)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at androidx.fragment.app.FragmentManager.dispatchPictureInPictureModeChanged(FragmentManager.java:3017)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at androidx.fragment.app.FragmentManager.lambda$new$3$androidx-fragment-app-FragmentManager(FragmentManager.java:468)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at androidx.fragment.app.FragmentManager$$ExternalSyntheticLambda3.accept(Unknown Source:4)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at androidx.activity.ComponentActivity.onPictureInPictureModeChanged(ComponentActivity.java:1070)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.app.Activity.dispatchPictureInPictureModeChanged(Activity.java:9320)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.app.ActivityThread.handleWindowingModeChangeIfNeeded(ActivityThread.java:6799)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.app.ActivityThread.performActivityConfigurationChanged(ActivityThread.java:6599)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.app.ActivityThread.performConfigurationChangedForActivity(ActivityThread.java:6574)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.app.ActivityThread.handleActivityConfigurationChanged(ActivityThread.java:6966)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.app.ActivityThread.handleActivityConfigurationChanged(ActivityThread.java:6899)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.app.servertransaction.ActivityConfigurationChangeItem.execute(ActivityConfigurationChangeItem.java:55)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:139)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:96)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2685)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:230)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:319)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:8919)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
06-14 15:07:04.065 32493 32493 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
</details>

Will try to setup a repro

Reply:
Does your production app contain 2e3b13b commit?

Reply:
yes

Reply:
I was not able to reproduce the issue with a minimal reproduction. I will dig in my app code, to see what go wrong

Reply:
@gkueny
You can easily avoid exception with a null check.
```diff
- ((ViewGroup)exoPlayerView.getParent()).removeView(exoPlayerView);
+ ViewGroup parent = (ViewGroup)exoPlayerView.getParent();
+ if (parent != null) {
+   parent.removeView(exoPlayerView);
+ }
```
But above code is not a proper fix and I don't know what side effects it will cause.
I'll see if I can reproduce it, but I don't think it will be easy.😰

Reply:
If this workaround, crash occur when I exit PiP 😅

In my app I reproduce the issue 100% with this flow:
- Show video
- Hide video
- Re-show video
- Go to background -> crash 

But with a simple reproduction cars do not happen 🧐
I deal with other issue so I cannot check deeper now, but will report to you if I find anything

Reply:
@gkueny 
Can you tell me the specific behavior of show and hide, e.g. is it mount/unmount, is it hidden by another view, etc.

Reply:
@YangJonghun , we display our stream in a global instance that is mount when a stream is show & unmount when no stream is show (normal or PiP). This global instance is displayed bwith absolute position (but I do not think it's matter)

Reply:
Minimal reproduction => https://github.com/gkueny/PictureInPictureAndroidCrashIssue

Reply:
@gkueny
fixed with [9c9fe4c](https://github.com/TheWidlarzGroup/react-native-video/pull/3385/commits/9c9fe4c7a2356fd0bfee14041c31100721984012)
Thanks for your help to investigation 👍